miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/EditRunController.java

```diff
@@ -312,4 +320,85 @@ public String processSubmit(@ModelAttribute("run") Run run, ModelMap model, Sess
       throw ex;
     }
   }
+
+  @Value("${miso.run.report.links:}")
+  private String runReportLinksConfigLine;
+
+  private Map<String, String> getRunReportServicesUrls(Run run) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 23 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)